### ISSUE TITLE
Does Lezer support (or plan to support) local variables?

### ISSUE THREAD

**Issue by jasonhan-vassar** [lezer-parser/lezer#41]
Hi, does Lezer support local variables like [tree-sitter does](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#local-variables)? In tree-sitter, I can declare local scopes and have one variable appear with a consistent style across the scope. Here is an example below. Note that the underlined variables appear in either a method argument/for loop declaration and remain underlined throughout the scope.

<img width="412" alt="Screen Shot 2020-12-09 at 6 14 19 PM" src="https://user-images.githubusercontent.com/34594853/101700237-b5fa4200-3a4a-11eb-8dc5-92ed3876dd39.png">

Thanks! I stumbled on your write-up on Lezer and since I already really love tree-sitter (moved to that recently; been stuck with tmLanguage/sublime-syntax for far too long), I really hope this project takes off :)

**Comment by marijnh** [comment]
No, it doesn't. Though from the tree-sitter docs, this seems to not really be a parser feature, but a highlighter one. Lezer node props could be used to annotate scope, variable definition, and variable reference nodes, and the highlighter could be extended to look at those (at the cost of doing some additional work).

I've been thinking about doing something like that for basic autocompletion and local jump-to-definition as well, though it seems that the result would still be rather crude (doesn't work for properties and across files) and probably not up to the expectations of people used to language-server-based completion.
